Mathlib/Data/Set/Pointwise/Interval.lean

```diff
@@ -60,53 +60,61 @@ end ContravariantLE
 section ContravariantLT
 
 variable [Mul α] [PartialOrder α]
-variable [CovariantClass α α (· * ·) (· < ·)] [CovariantClass α α (Function.swap HMul.hMul) LT.lt]
+variable [MulLeftStrictMono α] [MulRightStrictMono α]
 
 @[to_additive Icc_add_Ico_subset]
 theorem Icc_mul_Ico_subset' (a b c d : α) : Icc a b * Ico c d ⊆ Ico (a * c) (b * d) := by
-  haveI := covariantClass_le_of_lt
+  haveI := mulLeftMono_of_mulLeftStrictMono α
+  haveI := mulRightMono_of_mulRightStrictMono α
```

Comment:
Why do we need two instances instead of one? Also, again, `haveI` can probably be replaced by `have`. Same for everything below in this file.